lib/dataset/aligned_dict.cpp

```diff
@@ -280,38 +287,38 @@ Dict<Key, Value> &Dict<Key, Value>::setSlice(const Slice &s, const Dict &dict) {
 }
 
 template <class Key, class Value>
-void Dict<Key, Value>::rename(const Dim from, const Dim to) {
+void AlignedDict<Key, Value>::rename(const Dim from, const Dim to) {
   m_sizes.replace_key(from, to);
-  for (auto &item : m_items)
-    if (item.second.dims().contains(from))
-      item.second.rename(from, to);
+  for (auto it = m_items.values_begin(); it != m_items.values_end(); ++it)
+    if ((*it).dims().contains(from))
+      (*it).rename(from, to);
 }
```

Comment:
This comment is about the Python `rename` method, which is *not* the same as this (this method here is just doing the equivalent of `rename_dims`, which is also used to implemented `rename`):

https://github.com/scipp/scipp/blob/5f9d8256dfa6251808f5a7d226177fbda075dc52/src/scipp/core/dimensions.py#L132

does not preserve insertion order: The renamed coords will be moved to the end. Is that what we want to do? Or do we want to be able to rename keys without changing the order?

Reply:
As part of fixing #2734 I will be changing the C++ `rename` methods.

Reply:
I think we should preserve the order. We can implement it entirely in this class. But I'm going to wait for your changes.

Reply:
See #2775. I think the new implementation should automatically preserve order once the new dict is used.

Reply:
It doesn't because it didn't touch `rename`. We still pop a coord and append it at the end.